src/comm/pub_handler.cpp

```diff
@@ -177,6 +185,39 @@ void PubHandler::OnLivoxLidarPointCloudCallback(uint32_t handle, const uint8_t d
   return;
 }
 
+void
+PubHandler::OnLivoxLidarStateInfoCallback(const uint32_t handle,
+                                          const uint8_t dev_type,
+                                          const char* info,
+                                          void* client_data)
+{
+  PubHandler* self = (PubHandler*)client_data;
+  if (!self)
+  {
+    return;
+  }
+
+  if (self->state_info_callback_)
+  {
+    StateInfoData state_info;
+    state_info.lidar_type = static_cast<uint8_t>(LidarProtoType::kLivoxLidarType);
+    state_info.device_type = dev_type;  // corresponds to LivoxLidarDeviceType
+    state_info.handle = handle;
+    state_info.time_stamp = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+
+    state_info.info.Parse(info);
+    if (!state_info.info.IsObject())
+    {
+      std::cout << "OnLivoxLidarStateInfoCallback: Failed to parse the state_info json string" << std::endl;
```

Comment:
maybe use `std::cerr` - but i guess if they don't use it in the normal code, just stick to their coding style